client/src/components/MainFrame.js

```diff
@@ -93,30 +108,84 @@ function SimpleTable() {
       </Table>
   );
 }
+
 export default function(){
+  const classes = useStyles()
 	const [menuName, setMenuName]		= React.useState(/* default menu */'Verify')
+  const [isFilterShown, setFilterShown] = React.useState(false)
+  const [isMenuShown, setMenuShown] = React.useState(false)
 	const refContainer		= React.useRef()
 
 	function handleMenuClick(menuName){
 		setMenuName(menuName)
+    setMenuShown(false)
 	}
 
+  function handleToggleMenu(){
+    setMenuShown(!isMenuShown)
+  }
+
+  function handleFilterClick() {
+    if (isFilterShown) {
+      setFilterShown(false);
+    } else {
+      setFilterShown(true);
+    }
+  }
+
 	return(
-		<Grid container wrap='nowrap' >
+		<Grid container wrap='nowrap'>
 			<Grid item>
-        {/* hide menu 
-				<Menu
-					active={menuName}
-					onClick={handleMenuClick}
-				/>
-        */}
-			</Grid>
+        <AppBar
+          color='default'
+          className={classes.appBar}
+        >
+          <Grid container direction='column'>
+            <Grid item>
+              <Grid container justify='space-between'>
+                <Grid item>
+                  <IconButton onClick={handleToggleMenu}>
+                    <MenuIcon/>
+                  </IconButton>
+                  <IconLogo/>
+                </Grid>
+                <Grid item>
+                  <IconButton
+                    onClick={handleFilterClick}
+                  >
+                    <IconFilter />
+                  </IconButton>
+                </Grid>
+              </Grid>
+            </Grid>
+            {/*isFilterShown &&
```

Comment:
The filter was removed? How would it work then?

Reply:
Yeah that's the problem I'm having right now since it looks like we want the Menu bar on all the pages but only the filter button when we're viewing TreeImages. Was hoping for some advice maybe on how to navigate this issue! We could potentially create 2 separate top navbars, one for the TreeImages and another for everything else. 

For now I had just commented out the Filter part so that the page doesn't break when testing it out. Let me know if you'd like for me to keep digging into a solution!